src/common/chains.ts

```diff
@@ -9,3 +11,383 @@ export enum SupportedChainId {
   BASE = 8453,
   SEPOLIA = 11155111,
 }
+
+/**
+ * The chain id of the chain.
+ *
+ * TODO: Should we generalize it even more to allow non-EVM chains? We should probably revisit also the chain interface, and some other types.
+ */
+export type ChainId = number
+
+export interface WebUrl {
+  url: string
+  name: string
+}
+
+export type ChainRpcUrls = {
+  http: readonly string[]
+  webSocket?: readonly string[] | undefined
```

Comment:
What if only websocket?
Also, what about shortening it, it's still easy to understand IMO.

```suggestion
  http?: readonly string[] | undefined
  ws?: readonly string[] | undefined
```

There is probably a type to mark at least one is required, I think?

Reply:
I tried to use the same type as the one in `viem`

Reply:
I think they want to define always HTTP, and only sometimes was. I will just drop the redundant `undefined` in the optional field